core/src/main/scala/org/apache/spark/util/collection/OpenHashSet.scala

```diff
@@ -45,7 +45,7 @@ class OpenHashSet[@specialized(Long, Int) T: ClassTag](
     loadFactor: Double)
   extends Serializable {
 
-  require(initialCapacity <= (1 << 29), "Can't make capacity bigger than 2^29 elements")
+  require(initialCapacity <= (1 << 30), "Can't make capacity bigger than 2^30 elements")
```

Comment:
Ah right, because it chooses the next greater power of 2 as a capacity, so this limit sounds correct.

Still I'm not sure why the code had 1 << 29 then, and actually the old `POSITION_MAX` was `0x0EFFFFFF` not `0xEFFFFFFF`. It looks like it really should be `0x7FFFFFFF` but maybe we should check with @rxin to make sure I'm not missing something?